File: src/com/anathema_roguelike/entities/characters/foes/traits/Supernatural.java
package com.anathema_roguelike.entities.characters.foes.traits;

import com.anathema_roguelike.main.utilities.Utils;

public class Supernatural<T extends Attribute> extends StatTrait<T> {
	
	public Supernatural(Class<T> attribute) {
		super("Supernatural: " + Utils.getName(attribute), attribute, MultiplicativeCalculation.fixed(1.75));
	}
}

File: src/com/anathema_roguelike/entities/characters/foes/traits/Supernatural.scala
package com.anathema_roguelike
package entities.characters.foes.traits

import com.anathema_roguelike.main.utilities.Utils
import com.anathema_roguelike.stats.characterstats.attributes.Attribute
import com.anathema_roguelike.stats.effects.MultiplicativeCalculation
import scala.reflect.runtime.universe._

class Supernatural[T <: Attribute : TypeTag]()
  extends StatTrait[T]("Supernatural: " + Utils.getName(typeTagToClass[T]), MultiplicativeCalculation.fixed(1.75)) {
}